src/main/java/org/apache/commons/imaging/ImageParser.java

```diff
@@ -682,8 +688,10 @@ public abstract Dimension getImageSize(ByteSource byteSource, Map<String, Object
      *                            parser implementation.
      * @throws IOException        In the event of unsuccessful read or access operation.
      */
-    public abstract String getXmpXml(ByteSource byteSource, Map<String, Object> params)
-            throws ImageReadException, IOException;
+    public String getXmpXml(final ByteSource byteSource, final Map<String, Object> params)
```

Comment:
On 23.02.2015 20:15, Benedikt Ritter
      wrote:
    
    
      In src/main/java/org/apache/commons/imaging/ImageParser.java:
      > @@ -682,8 +688,10 @@ public abstract Dimension getImageSize(ByteSource byteSource, Map<String, Object
>       *                            parser implementation.
>       * @throws IOException        In the event of unsuccessful read or access operation.
>       */
> -    public abstract String getXmpXml(ByteSource byteSource, Map<String, Object> params)
> -            throws ImageReadException, IOException;
> +    public String getXmpXml(final ByteSource byteSource, final Map<String, Object> params)

      I'm not sure whether it is a good idea to have this method here
        if it doesn't make sense for all formats (as stated in the
        JavaDoc)
      —
        Reply to this email directly or view
          it on GitHub.
      
        
          
          
        
        
      
    
    I found an abstract method and two ways to implement it:
    a. return some String != null
    b. return null
    
    Of course I did not touch all implementations of type (a). It seems
    that no all formats supply xmp xml strings. I made this patch to
    remove the need for implementations of type (b). Now everyone knows:
    If my implementation can not return any xmp xml it should return
    null. This was my intention. If we need another default return value
    than null - no problem.
    -- 
Michael Groß